src/ng/document.js

```diff
@@ -30,3 +30,29 @@ function $DocumentProvider() {
     return jqLite(window.document);
   }];
 }
+
+
+/**
+ * @private
+ * Listens for document visibility change and makes the current status accessible.
+ */
+function $$IsDocumentHiddenProvider() {
+  this.$get = ['$document', '$rootScope', function($document, $rootScope) {
+    var doc = $document[0];
+    var hidden = doc && doc.hidden;
+
+    $document.on('visibilitychange', changeListener);
+
+    $rootScope.$on('$destroy', function() {
+      $document.off('visibilitychange', changeListener);
+    });
+
+    function changeListener() {
+      hidden = doc.hidden;
```

Comment:
I think the check was originally introduced by @matsko in case the document isn't available. I assume that when the document isn't available I can't add a listener anyway.